docs/en/operations/settings/settings.md

```diff
@@ -404,6 +404,38 @@ Possible values:
 
 Default value: 0.
 
+## partial_merge_join_optimizations {#partial_merge_join_optimizations}
+
+Enables of disables optimizations in partial merge join algorithm for [JOIN](../../sql-reference/statements/select/join.md) queries.
+
+Optimizations:
+
+- Skipping of  unneeded right blocks instead of pass through them.
+
+Possible values:
+
+-   0 — Optimizations disabled.
+-   1 — Optimizations enabled.
+
+Default value: 1.
+
+## partial_merge_join_rows_in_right_blocks {#partial_merge_join_rows_in_right_blocks}
+
+Limits a size of right-hand join data blocks.
+
+ClickHouse server:
+
+- Splits right-hand join data in blocks of the specified size.
+- Indexes each block with minimum and maximum values
+- Unloads prepared blocks to disk if possible.
+
+Possible values:
+
+- Any positive integer.
+- 0 — Limitation is turned off.
```

Comment:
Needs some clarification on what does it mean. The data is not split into blocks at all?

Reply:
It's not correct. It should be set to some adequate value: 1000 - 100000. It's possible to set it out of this range but it's not recomended. So `Any positive integer. Adequate values: 1000 - 100000. Recomended value: default`